components/server/src/user/user-controller.ts

```diff
@@ -600,16 +600,24 @@ export class UserController {
         req.query.returnTo = this.getSafeReturnToParam(req);
     }
 
+    protected urlStartsWith(url: string, prefixUrl: string): boolean {
+        prefixUrl += prefixUrl.endsWith("/") ? "" : "/";
+        return url.toLowerCase().startsWith(prefixUrl.toLowerCase());
+    }
+
     protected getSafeReturnToParam(req: express.Request) {
         const returnToURL: string | undefined = req.query.redirect || req.query.returnTo;
-        if (returnToURL) {
-            const hostUrl = this.env.hostUrl.url as URL;
-            if (returnToURL.toLowerCase().startsWith(`${hostUrl.protocol}//${hostUrl.host}`.toLowerCase())) {
-                return returnToURL;
-            }
-            if (returnToURL.toLowerCase().startsWith(this.env.brandingConfig.homepage.toLowerCase())) {
-                return returnToURL;
-            }
+        if (!returnToURL) {
+            log.debug({ sessionId: req.sessionID }, "Empty redirect URL");
+            return;
+        }
+
+        const hostUrl = this.env.hostUrl.url as URL;
```

Comment:
nit: you can remove this and directly call `this.env.hostUrl.toString()` below.

Reply:
Done ✔️